
Get rid of getbsize.  Default in upstream is 512, but we might want to have
something sane instead...

---
 sbin/swapon/Makefile |    2 +-
 sbin/swapon/swapon.c |   13 +++++++++++--
 2 files changed, 12 insertions(+), 3 deletions(-)

--- a/sbin/swapon/swapon.c
+++ b/sbin/swapon/swapon.c
@@ -41,6 +41,7 @@
 #include <sys/cdefs.h>
 __FBSDID("$FreeBSD$");
 
+#include <sys/swap.h>
 #include <sys/stat.h>
 #include <sys/param.h>
 #include <sys/sysctl.h>
@@ -71,6 +72,8 @@
 	int ch, doall;
 	int sflag = 0, lflag = 0, hflag = 0, qflag = 0;
 
+	setprogname (argv[0]);
+
 	if ((ptr = strrchr(argv[0], '/')) == NULL)
 		ptr = argv[0];
 	if (strstr(ptr, "swapon"))
@@ -80,7 +83,7 @@
 	orig_prog = which_prog;
 	
 	doall = 0;
-	while ((ch = getopt(argc, argv, "AadghklmqsU")) != -1) {
+	while ((ch = getopt(argc, argv, "AadghklmqsUve")) != -1) {
 		switch(ch) {
 		case 'A':
 			if (which_prog == SWAPCTL) {
@@ -132,6 +135,9 @@
 				usage();
 			}
 			break;
+		case 'v':
+		case 'e':
+			break;
 		case '?':
 		default:
 			usage();
@@ -271,7 +277,8 @@
 	    hlen = 10;
 	    break;
 	default:
-	    getbsize(&hlen, &blocksize);
+	    blocksize = 1024 * 1024;
+	    hlen = 10;
 	    snprintf(buf, sizeof(buf), "%ld-blocks", blocksize);
 	    break;
 	}
--- a/sbin/swapon/Makefile
+++ b/sbin/swapon/Makefile
@@ -9,6 +9,6 @@
 MLINKS+=swapon.8 swapctl.8
 
 DPADD=	${LIBUTIL}
-LDADD=	-lutil
+LDADD=	-lbsd -lutil
 
 .include <bsd.prog.mk>
